File: practica2_839756_842236/c++/instruccionAdd.cc
// Autores: Andrei Vlasceanu [839756] & Andres Yubero [842236]

#include "instruccionAdd.h"

void InstruccionAdd::ejecutar(stack<int> *s)
{
    int x, y, res;
    x = s->top();
    s->pop();
    y = s->top();
    s->pop();
    res = x + y;
    s->push(res);
}

void InstruccionAdd::listar()
{
    cout << "add" << endl;
}

File: practica2_839756_842236/java/InstruccionAdd.java
import java.util.Stack;

public class InstruccionAdd extends Instruccion {
    @Override
    public void ejecutar(Stack<Integer> s) {
        int x = s.pop();
        int y = s.pop();
        s.push(x + y);
    }

    @Override
    public void listar() {
        System.out.println("add");
    }
}
